db/migrate/20191023180308_subclass_ems_cluster.rb

```diff
@@ -0,0 +1,41 @@
+class SubclassEmsCluster < ActiveRecord::Migration[5.1]
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class EmsCluster < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    [
+      'Kubevirt',
+      'Openstack',
+      'Microsoft',
+      'Redhat',
+      'Vmware',
+    ].each do |provider|
+      ems_class_name = "ManageIQ::Providers::#{provider}::InfraManager"
+      cluster_class_name = "#{ems_class_name}::Cluster"
+      EmsCluster.in_my_region.
+        joins('join ext_management_systems on ems_clusters.ems_id = ext_management_systems.id').
+        where(:ext_management_systems => {:type => ems_class_name}).
+        update_all(:type => cluster_class_name)
+    end
+  end
+
+  def down
+    # Set back all the clusters to plain old EmsCluster
+    EmsCluster.in_my_region.
+      where.not(:type => 'ManageIQ::Providers::Openstack::InfraManager::EmsCluster').
```

Comment:
After the `#up` is run all OpenStack clusters will be `::InfraManager::Cluster` not `::InfraManager::EmsCluster`.  This is essentially setting all clusters to `:type => nil` because no clusters will have this type when the down is run.